core/services/relay/evm/cap_encoder.go

```diff
@@ -84,41 +84,60 @@ func (c *capEncoder) Encode(ctx context.Context, input values.Map) ([]byte, erro
 }
 
 func prependMetadataFields(meta consensustypes.Metadata, userPayload []byte) ([]byte, error) {
-	// TODO: use all 7 fields from Metadata struct
-	result := []byte{}
-	workflowID, err := decodeID(meta.WorkflowID, idLen)
-	if err != nil {
+	var err error
+	var result []byte
+
+	// 1. Version (1 byte)
+	if meta.Version > 255 {
```

Comment:
Could this just be typed as `uint8`/`byte` so we don't need the check here?

Reply:
I think values.Map had some issues with uint8 but let me check again.

Reply:
I'll leave it as uint32 for simpler value encoding.